### DISCUSSION TITLE
How to set which webcam to use in WebCamConfig

### DISCUSSION THREAD

**Discussion by IgorSamer** [vladmandic/human#311] [Q&A] [answered]
I'm using the following code to init the webcam:

```
await human.webcam.start({ crop: true })

human.video(human.webcam.element)

canvas.width = human.webcam.width
canvas.height = human.webcam.height

drawLoop()
```

But when running on notebooks with two webcams (built-in and external) I can't define which webcam I want to use, in some cases even having to disable the built-in webcam for Human be able to detect the external one instead. Is there any way to define the camera label I want to use? I ended up not finding it in the documentation.

I would also like to know what the `crop: true` property does exactly, as in the docs it just says "camera crop mode" and it wasn't very clear to me.

Thanks!

**Comment by vladmandic** [comment, marked as answer]
> Is there any way to define the camera label I want to use  

not explicitly, but its a good feature-request, i'll add it.  

in the meantime, you can try `mode: front` or `mode: back` if it helps.  
(that api is not explictly designed to select multiple webcams, but for mobile phones with user-facing or back-facing cameras.)  

> I would also like to know what the crop: true property does exactly  

`crop: true` means when width/height are set, it will perform a center-crop to exact specified width/height.  
if `crop: false`, it will select stream that fits contstraints, but if it doesn't match width/height exactly, it will leave it as-is.  

for example, you may want **500 x 500**, but camera does not have that natively, so it may select **1024 x 768** as closest native resolution  
so do you want to keep it as native resolution or as cropped?

**Reply by IgorSamer** [reply]
> not explicitly, but its a good feature-request, i'll add it.

It would be perfect!

> crop: true means when width/height are set, it will perform a center-crop to exact specified width/height.
> if crop: false, it will select stream that fits contstraints, but if it doesn't match width/height exactly, it will leave it as-is.

Got it! So there was no need to use `crop: true` in my case as I didn't set the width and height either.

**Comment by vladmandic** [comment]
added option to specify which video source to use in `human.webcam`  

example:
```js
const devices = await human.webcam.enumerate();
const id = devices[0].deviceId; // use first available video source
await human.webcam.start({ element: videoElement, crop: true, width: 800, id });
```

code is part of upcoming `human` release v3, comitted in github main branch, but not yet released on npmjs.

**Reply by IgorSamer** [reply]
Amazing! Thank you so much!
